### PR TITLE
feat: Input and Label primitives

### PR DESCRIPTION
**Pull request by reesscot**
#### Description of changes
Export `Input` and `Label` primitives for external use. Docs:
https://reesscot-add-input.dvmvffzts1tcu.amplifyapp.com/react/components/input
https://reesscot-add-input.dvmvffzts1tcu.amplifyapp.com/react/components/label

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/ui/src/theme/createTheme.ts [hunk 1]
```diff
@@ -98,9 +98,15 @@ export function createTheme(
       }
       if ('colorMode' in override) {
         cssText += `\n@media (prefers-color-scheme: ${override.colorMode}) {
-          [data-amplify-theme="${name}"][data-amplify-color-mode="system"] {\n${customProperties}\n}
+          [data-amplify-theme="${name}"][data-amplify-color-mode="system"] {
+            ${customProperties}
+            color-scheme: ${override.colorMode};
```
Comment (reesscot) [inline packages/ui/src/theme/createTheme.ts:103]: Fixes an issue where Chrome doesn't pickup the system color theme, and the "date" button is the wrong color.

Before:
CleanShot 2023-08-28 at 17 24 17@2x

After:

File: docs/__tests__/__snapshots__/sitemap.test.ts.snap [hunk 2]
```diff
@@ -68,6 +68,7 @@ exports[`Sitemap Snapshot 1`] = `
 /react/components/highlightmatch, 
 /react/components/icon, 
 /react/components/image, 
+/react/components/input,
```
Comment (dbanksdesign) [inline docs/__tests__/__snapshots__/sitemap.test.ts.snap:71]: No label page on the sitemap?

File: docs/src/data/links.tsx [hunk 3]
```diff
@@ -288,7 +303,7 @@ export const inputComponents: ComponentNavItem[] = [
   {
     href: '/components/textfield',
     label: 'Text Field',
-    body: `The TextField form primitive can be used allow users to input text content.`,
+    body: `The TextField form primitive allows users to input text content.`,
```
Comment (dbanksdesign) [inline docs/src/data/links.tsx:306]: Good catch

File: docs/src/pages/[platform]/components/input/InputPropControls.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import {
+  InputProps,
+  TextField,
+  SelectField,
+  SwitchField,
+  Flex,
+} from '@aws-amplify/ui-react';
+import * as React from 'react';
+
+export interface InputPropControlsProps extends InputProps {
```
Comment (dbanksdesign) [inline docs/src/pages/[platform]/components/input/InputPropControls.tsx:10]: Just thinkin out loud, nothing to do with this PR, all this stuff should be automatically generated

File: docs/src/pages/[platform]/components/input/examples/DefaultRequiredInputExample.tsx [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Button, Flex, Input, Label } from '@aws-amplify/ui-react';
+
+export const DefaultRequiredInputExample = () => {
+  return (
+    <Flex as="form" direction="column" width="20rem">
+      <Label htmlFor="email">Email</Label>
+      <Input id="email" type="email" isRequired={true} />
```
Comment (dbanksdesign) [inline docs/src/pages/[platform]/components/input/examples/DefaultRequiredInputExample.tsx:7]: Could this just be
```suggestion
      <Input id="email" type="email" isRequired />
```

File: docs/src/pages/[platform]/components/input/examples/RequiredInputExample.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { Button, Flex, Text, Input, Label } from '@aws-amplify/ui-react';
+
+export const RequiredInputExample = () => {
+  return (
+    <Flex as="form" direction="column" width="20rem">
```
Comment (dbanksdesign) [inline docs/src/pages/[platform]/components/input/examples/RequiredInputExample.tsx:5]:
```suggestion
    <Flex as="form" direction="column" width="100%">
```

File: docs/src/pages/[platform]/components/input/examples/RequiredInputExample.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+      <Label htmlFor="email">
+        Email
+        <Text as="span" fontSize="0.8rem" color="red">
```
Comment (dbanksdesign) [inline docs/src/pages/[platform]/components/input/examples/RequiredInputExample.tsx:8]:
```suggestion
        <Text as="span" fontSize="small" color="font.error">
```

File: docs/src/pages/[platform]/components/input/react.mdx [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+      id="quantity"
+      type="number"
+    />
+    ```
+
+    </ExampleCode>
+
+  </Example>
+
+## CSS Styling
```
Comment (dbanksdesign) [inline docs/src/pages/[platform]/components/input/react.mdx:179]:
```suggestion
## Styling
```

**Review by dbanksdesign — APPROVED** [review]
Some tiny things. Preemptively approving

File: docs/src/pages/[platform]/components/input/InputPropControls.tsx (hunk 4 above)
Comment (reesscot) [reply docs/src/pages/[platform]/components/input/InputPropControls.tsx:10]: 100%

File: docs/__tests__/__snapshots__/sitemap.test.ts.snap (hunk 2 above)
Comment (reesscot) [reply docs/__tests__/__snapshots__/sitemap.test.ts.snap:71]: Fixed

**Review by dbanksdesign — DISMISSED** [review]
LGTM!

**Review by hbuchel — DISMISSED** [review]

**Review by dbanksdesign — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

**Pull request merged**